pywr-schema/src/timeseries/mod.rs

```diff
@@ -0,0 +1,153 @@
+mod polars_dataset;
+
+use ndarray::{Array1, Array2};
+use polars::error::PolarsError;
+use polars::prelude::DataType::Float64;
+use polars::prelude::{DataFrame, Float64Type, IndexOrder, Schema};
+use pywr_core::models::ModelDomain;
+use pywr_core::parameters::{Array1Parameter, Array2Parameter, ParameterIndex};
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+use thiserror::Error;
+
+use crate::{parameters::ParameterMeta, SchemaError};
+
+use self::polars_dataset::PolarsDataset;
+
+#[derive(serde::Deserialize, serde::Serialize, Debug, Clone)]
+#[serde(tag = "type")]
+enum TimeseriesProvider {
+    Pandas,
+    Polars(PolarsDataset),
+}
+
+struct ResamplingArgs {}
+
+#[derive(serde::Deserialize, serde::Serialize, Debug, Clone)]
+pub struct Timeseries {
+    #[serde(flatten)]
+    names: ParameterMeta,
+    provider: TimeseriesProvider,
+    url: PathBuf,
+}
+
+impl Timeseries {
+    pub fn load(&self, domain: &ModelDomain, data_path: Option<&Path>) -> Result<DataFrame, SchemaError> {
+        match &self.provider {
+            TimeseriesProvider::Polars(dataset) => dataset.load(self.url.as_path(), data_path),
+            TimeseriesProvider::Pandas => todo!(),
+        }
+    }
+}
+
+fn align_and_resample(df: &DataFrame, domain: &ModelDomain) -> Result<DataFrame, SchemaError> {
```

Comment:
This is what I imagined this might look like. Perhaps it returns an `Array2` instead of `DataFrame`?

I think we might want this in a separate module with a bunch of separate unit tests. 

Also, I guess there's some convention about the format / structure of a time column in `df`?